File: Sliding Window/leetcode_1423.py

# Problem link ===>>  https://leetcode.com/problems/maximum-points-you-can-obtain-from-cards/description/
# submission link == =>> https://leetcode.com/problems/maximum-points-you-can-obtain-from-cards/submissions/1590253312/

# from typing import List
class List:
    def __init__(self):
        self.data = []

    def append(self, item):
        self.data.append(item)


class Solution:
    def maxScore(self, C: List[int], K: int) -> int:
        best = total = sum(C[:K])
        for i in range(K-1, -1, -1):
            total += C[i + len(C) - K] - C[i]
            best = max(best, total)
        return best


if __name__ == "__main__":
    solution = Solution()

File: Sliding Window/leetcode_1423.cpp

// Problem link ===>>  https://leetcode.com/problems/maximum-points-you-can-obtain-from-cards/description/

#include <bits/stdc++.h>
using namespace std;
#define ll long long
#define endl '\n'

class Solution
{
public:
    int maxScore(vector<int> &C, int K)
    {
        int total = 0;
        for (int i = 0; i < K; i++)
        {
            total += C[i];
        }
        int best = total;
        for (int i = K - 1, j = C.size() - 1; ~i, i--, j--;)
        {
            total += C[j] - C[i];
            best = max(best, total);
        }
    }
};

int main()
{
    ios_base::sync_with_stdio(false);
    cout.tie(nullptr);
    cin.tie(nullptr);

    ll t;
    cin >> t;
    while (t--)
    {
        ll n;
        cin >> n;
    }
    return 0;
}